### DISCUSSION TITLE
Inverted commas

### DISCUSSION THREAD

**Discussion by smeech** [espanso/espanso#1451] [Q&A] [answered]
Why does the Espanso documentation include the use of inverted commas for parameters when in most cases (`trigger:`, `label:`, `id:`), except `replace:`, they don't appear necessary even when the parameters contain spaces?

**Comment by nudefireninja** [comment, marked as answer]
The configuration format is YAML and that has pretty intricate [rules for quoting](https://www.yaml.info/learn/quote.html), and when it is optional versus required. When you omit the quotes (or inverted commas as you call them) it's called a "plain" flow scalar and that works fine in many cases, but can be misinterpreted in others.

The safest option is to always use quotation marks for text (either double- or single-quotes), but leave numbers and `true`/`false` values plain. This will result in better compatibility with old versions of YAML and a better syntax highlighting experience. In espanso specifically, I would always quote the `trigger` and `replace` values for the sake of consistency.
